services/src/main/scala/cromwell/services/cost/GcpCostCatalogService.scala

```diff
@@ -212,23 +235,47 @@ class GcpCostCatalogService(serviceConfig: Config, globalConfig: Config, service
     }
 
   // TODO consider caching this, answers won't change until we reload the SKUs
-  def calculateVmCostPerHour(instantiatedVmInfo: InstantiatedVmInfo): ErrorOr[BigDecimal] =
-    for {
+  def calculateVmCostPerHour(instantiatedVmInfo: InstantiatedVmInfo): ErrorOr[BigDecimal] = {
+    val cpuPricingInfoErrorOr = for {
       cpuSku <- lookUpSku(instantiatedVmInfo, Cpu)
       coreCount <- MachineType.extractCoreCountFromMachineTypeString(instantiatedVmInfo.machineType)
       cpuPricePerHour <- GcpCostCatalogService.calculateCpuPricePerHour(cpuSku, coreCount)
+    } yield (cpuSku, coreCount, cpuPricePerHour)
+
+    val ramPricingInfoErrorOr = for {
       ramSku <- lookUpSku(instantiatedVmInfo, Ram)
       ramMbCount <- MachineType.extractRamMbFromMachineTypeString(instantiatedVmInfo.machineType)
       ramGbCount = ramMbCount / 1024d // need sub-integer resolution
       ramPricePerHour <- GcpCostCatalogService.calculateRamPricePerHour(ramSku, ramGbCount)
-      totalCost = cpuPricePerHour + ramPricePerHour
+    } yield (ramSku, ramGbCount, ramPricePerHour)
+
+    val gpuPricingInfoErrorOr = instantiatedVmInfo.gpuInfo match {
+      case None => (None, 0, BigDecimal(0)).validNel
+      case Some(gpuInfo) =>
+        for {
+          gpuSku <- lookUpSku(instantiatedVmInfo, Gpu)
+          gpuCount = gpuInfo.count
+          gpuPricePerHour <- GcpCostCatalogService.calculateGpuPricePerHour(gpuSku, gpuCount)
+        } yield (Some(gpuSku), gpuCount, gpuPricePerHour)
+    }
+
+    for {
+      cpuPricingInfo <- cpuPricingInfoErrorOr
+      (cpuSku, coreCount, cpuPricePerHour) = cpuPricingInfo
+      ramPricingInfo <- ramPricingInfoErrorOr
+      (ramSku, ramGbCount, ramPricePerHour) = ramPricingInfo
+      gpuPricingInfo <- gpuPricingInfoErrorOr
+      (gpuSku, gpuCount, gpuPricePerHour) = gpuPricingInfo
+      totalCost = cpuPricePerHour + ramPricePerHour + gpuPricePerHour
       _ = logger.info(
         s"Calculated vmCostPerHour of ${totalCost} " +
           s"(CPU ${cpuPricePerHour} for ${coreCount} cores [${cpuSku.getDescription}], " +
-          s"RAM ${ramPricePerHour} for ${ramGbCount} Gb [${ramSku.getDescription}]) " +
+          s"RAM ${ramPricePerHour} for ${ramGbCount} Gb [${ramSku.getDescription}], " +
+          s"GPU ${gpuPricePerHour} for ${gpuCount} GPUs [${gpuSku.map(_.getDescription)}]) " +
```

Comment:
Unlike with the CPU and RAM SKUs, I sent back the GPU SKU from the GPU cost `for-yield` as an `Option[Sku]` since we don't always have a GPU SKU, and I used the Option in the log message. We should try to make this more readable.